libraries/redcore/api/hal/hal.php

```diff
@@ -662,13 +669,14 @@ public function apiCreate()
 		}
 
 		$this->setData('result', $result);
-		$this->triggerFunction('displayErrors', $model);
+		$this->triggerFunction('displayErrors', $model->getErrors());
```

Comment:
this is wrong, you should just provide a model like it was before, it will already execute getErrors if the method exists in that class...